tests/cases/conformance/jsdoc/jsdocStringLiteral.ts

```diff
@@ -0,0 +1,12 @@
+// @allowJs: true
+// @filename: in.js
+// @out: out.js
+/**
+ * @param {'literal'} p1
+ * @param {"literal"} p2
+ * @param {'literal' | 'other'} p3
+ * @param {'literal' | number} p4
+ */
+function f(p1, p2, p3, p4) {
+    return p1 + p2 + p3 + p4 + '.';
```

Comment:
Does this test really do anything apart from test baselines? Why not write some tests that have observable characteristics from a usage standpoint?

For instance, string completion when calling something that takes a union type of string literals.